src/services/UsersService.ts

```diff
@@ -82,21 +82,39 @@ export class UsersService {
   }
 
   async cancelUserIndividualSubscriptions(customerId: User['customerId']): Promise<void> {
+    if (!customerId) throw new Error('customerId required');
     const activeSubscriptions = await this.paymentService.getActiveSubscriptions(customerId);
 
     if (activeSubscriptions.length === 0) {
       throw new Error('Subscriptions not found');
     }
 
     const individualSubscriptions = activeSubscriptions.filter(
-      (subscription) => subscription.metadata.is_teams !== '1',
+      (subscription) => subscription.metadata.is_teams !== '1' && subscription.product?.metadata.type !== 'business',
```

Comment:
Remove the `is_teams` flag usage, we only use our current flags, not legacy ones